cogeo_mosaic/backend/base.py

```diff
@@ -0,0 +1,31 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+import contextlib
+from typing import Dict
+
+
+class BaseBackend(metaclass=contextlib.AbstractContextManager):
```

Comment:
I'm not sure if this is correct. https://github.com/rafelafrance/traiter_vertnet/blob/fc95fbdff9ebc7ab569d24f7b3e5ee644d2c6ad6/vertnet/writers/base_writer.py#L4-L7
does it slightly differently.

I got this code from https://pymotw.com/3/abc/

Reply:
I also do it differently https://github.com/vincentsarago/rio-tiler-experiment/blob/master/cog_tiler/base.py#L14 (mine is definitely worse). Not sure what is the best

Reply:
It wasn't working with `metaclass` so I switched to the use in my first link.